src/EditorFeatures/CSharpTest/UsePatternCombinators/CSharpUsePatternCombinatorsDiagnosticAnalyzerTests.cs

```diff
@@ -117,7 +118,9 @@ public async Task TestMissingOnExpression(string expression)
         [InlineData("!(i != 1 && 2 != i)", "i is 1 or 2")]
         [InlineData("i < 1 && 2 <= i", "i is < 1 and >= 2")]
         [InlineData("i < 1 && 2 <= i && i is not 0", "i is < 1 and >= 2 and not 0")]
-        [InlineData("(int.MaxValue - 1D) < i && i > 0", "i is > (int.MaxValue - 1D) and > 0")]
+        [InlineData("(int.MaxValue - 1D) < i && i > 0", "(double)i is > (int.MaxValue - 1D) and > 0")]
+        [InlineData("ch < ' ' || ch >= 0x100 || 'a' == ch", "(int)ch is < ' ' or >= 0x100 or 'a'")]
+        [InlineData("ch == 'a' || 'b' == ch", "(int)ch is 'a' or 'b'")]
```

Comment:
@CyrusNajmabadi For some reason, characters are converted to `int` even in equality comparison. Could be IOp bug?

Reply:
I think it is because no built-in `==` operator which compares `char`, `byte`, `short`, exists. https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#integer-comparison-operators

Reply:
So I guess we'll have to special case it here.

Reply:
To not duplicate logic I'm just going to attach the simplifier and rely on https://github.com/dotnet/roslyn/issues/45925 to fix it up.

Reply:
Alternatively we could cast the pattern itself, but it would still suffer from the same issue (unexpected conversion nodes are present in the tree).